src/relion/zocalo/cryolo.py

```diff
@@ -218,9 +219,46 @@ def dummy(self, *args, **kwargs):
                 np.array(cbox_block.find_loop("_Confidence"), dtype=float),
                 np.array(cbox_block.find_loop("_Confidence"), dtype=float),
             )
-            cryolo_particle_sizes = cbox_sizes[
-                cbox_confidence > cryolo_params.threshold
-            ]
+
+            # Select only a fraction of particles based on confidence if requested
+            if (
+                cryolo_params.retained_fraction < 1
+                and cryolo_params.retained_fraction > 0
+            ):
```

Comment:
Would it make sense to do this validation within the `pydanitc` model?

Reply:
Pydantic could be used to check if this is a valid fraction, yes.
The aim of the if statement here is to check if the threshold needs recalculating to get a chosen fraction of the particles though.